### PR TITLE
feat(findings): add rule catalog and platform docs

### PR DESCRIPTION
**Pull request by jonathanhaaswriter**
## Summary
- add a first-class finding rule catalog over HTTP and ConnectRPC so platform clients can discover supported `rule_id` values
- document the findings platform architecture, the why behind the rule registry, and quickstart usage for evaluation and reads
- add code-level comments and tests so the catalog and docs stay aligned with the current bootstrap findings surface

## Validation
- go -C /Users/jonathan/Documents/cerebro-worktrees/cerebro-next-okta-source-20260423 test ./internal/findings ./internal/bootstrap ./internal/statestore/postgres ./internal/reports
- make verify

### CODE REVIEW

**Comment by jonathanhaaswriter** [comment]
@droid Please perform a deep code review and security review of this PR. Leave inline comments only for high-confidence, actionable issues.

**Comment by jonathanhaaswriter** [comment]
@droid Please re-review after the latest pushed fixes. Focus on whether the prior findings on this PR are resolved and only report remaining high-confidence issues.

File: internal/bootstrap/app.go [hunk 1]
```diff
@@ -573,11 +632,45 @@ func (s *bootstrapService) ListFindings(ctx context.Context, req *connect.Reques
...
+
+func (s *bootstrapService) GetFindingEvaluationRun(ctx context.Context, req *connect.Request[cerebrov1.GetFindingEvaluationRunRequest]) (*connect.Response[cerebrov1.GetFindingEvaluationRunResponse], error) {
+	run, err := findings.New(
+		sourceRuntimeStore(s.deps.StateStore),
+		eventReplayer(s.deps.AppendLog),
+		findingStore(s.deps.StateStore),
+		findingEvaluationRunStore(s.deps.StateStore),
+	).GetEvaluationRun(ctx, req.Msg.GetId())
+	if err != nil {
+		return nil, err
```
Comment (josh-writer) [inline internal/bootstrap/app.go]: GetFindingEvaluationRun returns not-found as a plain error. ports.ErrFindingEvaluationRunNotFound becomes Connect CodeUnknown/HTTP 500 instead of NotFound, while the HTTP route maps the same error to 404.

File: internal/statestore/postgres/findingevaluationruns.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	trimmed := strings.TrimSpace(value)
+	if trimmed == "" {
+		return
+	}
+	*args = append(*args, trimmed)
+	*clauses = append(*clauses, fmt.Sprintf("%s = $%d", column, len(*args)))
+}
+
+func findingEvaluationRunTime(value interface{ AsTime() time.Time }) time.Time {
+	if value == nil {
```
Comment (josh-writer) [inline internal/statestore/postgres/findingevaluationruns.go:207]: findingEvaluationRunTime misses typed-nil protobuf timestamps. The initial “running” run has FinishedAt == nil, but passing run.GetFinishedAt() into an interface makes the nil check fail; AsTime() returns Unix epoch, so finished_at is stored non-NULL for unfinished runs.

**Review by josh-writer — CHANGES_REQUESTED** [review]

**Comment by jonathanhaaswriter** [comment]
### Droid sweep status

Latest sweep: `2026-04-29 22:09 UTC`.

- Unresolved non-outdated review threads: `0`
- CI `verify`: `SUCCESS`
- `droid-review`: `not present`

I’ll keep watching for new review findings/check failures and address them as they appear.

**Comment by jonathanhaaswriter** [comment]
Droid sweep audit complete (final pass)

- Re-scanned all live review threads, including outdated ones
- Verified each remaining thread against the current branch tip
- Resolved every actionable thread (or fixed it inline and pushed)
- 0 unresolved review threads across the whole open-PR stack at this time

PR #494 (fix/feedback-main-followup) and PR #495 (feat/droid-creation-workflows) received additional fixes for SSRF (DNS-rebind) and workflow branch-name validation in this pass.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid please harden this PR end-to-end:

1. Re-review the diff for residual bugs, races, edge cases, security gaps, and untested paths.
2. Apply the smallest correct fixes for any high-confidence issues you find.
3. Strengthen test coverage: add regression tests for every prior review finding on this PR plus the new behavior, including failure paths, boundary inputs, concurrency, and not-found / typed-nil interface guards.
4. Run focused tests and `make verify`. If validators fail, fix until green.
5. Commit and push the fixes to this PR's branch (or, if you lack push permission in this workflow, leave concrete inline suggestions and a summary comment).

Do not change the PR scope, do not modify the default branch, and do not merge.

**Comment by jonathanhaaswriter** [comment]
@droid review

**Comment by jonathanhaaswriter** [comment]
@droid review

Please re-review this rule-catalog PR deeply. Focus on subtle correctness/security issues:

1. **Rule identity and versioning**: check catalog IDs, slugs, fingerprints, migrations, and update semantics so rules do not collide or drift across deployments.
2. **Tenant/platform boundaries**: ensure platform/global rules cannot be accidentally shadowed, mutated, or read as tenant-owned state unless explicitly intended.
3. **Evaluation safety**: look for untrusted rule metadata flowing into SQL, CEL/expression evaluation, templates, logs, reports, or UI output without validation/escaping.
4. **Determinism**: verify catalog ordering, default severities, enablement flags, and matching behavior are deterministic across Go maps, DB reads, and generated docs/contracts.
5. **Compatibility with downstream finding PRs**: confirm rule catalog outputs line up with finding reads/evidence/evaluation runs and do not break existing persisted findings.
6. **Operational failure modes**: check missing/invalid catalog entries, partial migrations, duplicate rules, and whether errors are surfaced safely without panics or noisy internal details.

**Comment by jonathanhaaswriter** [comment]
Closing this PR — its substantive feature work is already in `main` via the earlier consolidation rollups, and the remaining unresolved factory-droid findings on this stack have been addressed in main via the follow-up PR:

- #500 (`fix(findings): persist normalized event limit and disambiguate run/evidence ids`)

PR #500 specifically pulls forward:

- Persisting the **normalized** event limit on `FindingEvaluationRun` records (PR #477)
- Hashing raw runtime/rule ids and appending a crypto-rand suffix to `findingEvaluationRunID` so concurrent multi-rule evaluations and normalization-equivalent ids never collide (PR #479)
- Appending a sha256 suffix to `findingEvidenceID` so lossy normalization no longer overwrites distinct evidence rows (PR #479)

This branch has diverged thousands of lines from `main` and rebasing it cleanly is no longer practical; closing here keeps the open queue accurate. If anything from this stack still needs to land, please open a fresh PR cut from `main`.
